### DISCUSSION TITLE
[Reanimated] Property `opacity` was whitelisted both as UI and native prop.

### DISCUSSION THREAD

**Discussion by dafrica1** [software-mansion/react-native-reanimated#5621] [Q&A]
Running into an odd issue while upgrading my Expo native project from Expo 49 to 50.

When attempting to run the app, I get the following error:
```
[Reanimated] Property `opacity` was whitelisted both as UI and native prop.
```
When I check the `react-native-reanimated` folder in `node_modules` and open the `PropsAllowlists.ts` file, I do not see opacity in both UI and native props, it's only in Native.

Why do I continue to get this error message?

**Comment by ambikaKum** [comment]
have you got the answer, we got the similar error

Property `stroke` was whitelisted both as UI and native prop. Please remove it from one of the lists.

**Reply by valeriavodianchuk** [reply]
Any updates?

**Reply by Kashifmub205** [reply]
This error occurs because `react-native-reanimated` detects `stroke` being registered as both a UI and native prop, causing a conflict. To fix this, remove `stroke` from direct props and move it inside `useAnimatedProps`.

```
<AnimatedCircle
  cx="50%"
  cy="100%"
  r={radius}
  strokeWidth={strokeWidth}
  fill="transparent"
  strokeDasharray={circumference}
  animatedProps={animatedProps} // No direct stroke prop
  strokeLinecap="round"
/>;

```
This ensures `stroke` is managed within `useAnimatedProps` resolving the conflict.

**Comment by MatiPl01** [comment]
Hey!
Do you guys use `Animated.addWhitelistedUIProps` or `Animated.addWhitelistedNativeProps` anywhere in your project? Can you provide your package.json so that I can see which libraries you use that may cause the problem and which version of `react-native-reanimated` you have in your project?

**Reply by seanblonien** [reply]
this is the one I use that is giving me the error for "stroke" property
https://github.com/nithinpp69/react-native-circular-progress-indicator

**Reply by MatiPl01** [reply]
Can you tell me which version of reanimated do you use and provide a code snippet which causes this crash?

**Reply by seanblonien** [reply]
this is the exact props i am passing to the underling library component which uses `react-native-reanimated`
```tsx
import {default as CircularProgress} from 'react-native-circular-progress-indicator';

const ProgressComponent: React.FC = () => {
  const value = 50;

  return (
    <CircularProgress
      activeStrokeWidth={12}
      duration={1000}
      progressFormatter={(value: number) => {
        'worklet';
        if (value === 100) return '100';
        
        return value.toFixed(1);
      }}
      progressValueColor='black'
      value={value}
      valueSuffix='%'
    />
  )
}
```
with these package version numbers
```json
"react-native-circular-progress-indicator": "4.4.2",
"react-native-reanimated": "3.6.3"
```

**Reply by dereckquock** [reply]
we're running into the same issue with [`react-native-circular-progress-indicator`](https://github.com/nithinpp69/react-native-circular-progress-indicator)

it might be an issue in `circular-progress-indicator` since that package hasn't been updated in 2 years... 😓

**Reply by JacktheGodzillaSlayer** [reply]
> we're running into the same issue with [`react-native-circular-progress-indicator`](https://github.com/nithinpp69/react-native-circular-progress-indicator)
> 
> it might be an issue in `circular-progress-indicator` since that package hasn't been updated in 2 years... 😓

That's true, we already experienced many issues regarding the old libraries which were out of maintainance conflicted with upgrade RN and SDK, the only way is to fix the library yourself and then use package-patch to do post npm actions

**Comment by dafrica1** [comment]
As a long-overdue follow up, the other component which is causing the conflict is this one: https://github.com/gorhom/react-native-bottom-sheet

It's a pretty well known one.

**Reply by MatiPl01** [reply]
Hey! Can you tell me which versions of `@gorhom/bottom-sheet` and `react-native-reanimated` libraries do you use?

I can see that the bottom sheet library specifies only one property as a whitelisted UI prop:

```tsx
Animated.addWhitelistedUIProps({
  decelerationRate: true,
});
```

This prop isn't mentioned in `react-native-reanimated` [allow list](https://github.com/software-mansion/react-native-reanimated/blob/main/packages/react-native-reanimated/src/propsAllowlists.ts), so maybe the other library is conflicting with the `@gorhom/bottom-sheet`. Would you be able to share your package.json and code snippet that causes this error?

You said it's a pretty well known issue. Can you provide a link to similar issue reports? I haven't found any similar issue report in the `@gorhom/bottom-sheet` library repository.

**Reply by dafrica1** [reply]
It's strange. You are right that `bottom-sheet` doesn't have `opacity` whitelisted but even when I search my node_modules folder, I can't find any other reference to reanimated.

There is only one instance to `addWhitelistedNative` in my code. The reason I say well known is due to several others commenting here about having the same issue.

My full package.json:
```
{
  "name": "my_app",
  "version": "1.0.0",
  "scripts": {
    "start": "expo start --dev-client",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web",
    "postinstall": "patch-package"
  },
  "dependencies": {
    "@amplitude/analytics-react-native": "^1.4.0",
    "@gorhom/bottom-sheet": "^4.4.6",
    "@gorhom/portal": "^1.0.14",
    "@react-native-async-storage/async-storage": "1.23.1",
    "@react-native-community/datetimepicker": "8.0.1",
    "@react-native-firebase/analytics": "^20.3.0",
    "@react-native-firebase/app": "^20.3.0",
    "@react-navigation/native": "^6.1.1",
    "@react-navigation/native-stack": "^6.9.7",
    "@sentry/react-native": "~5.22.0",
    "axios": "^1.7.3",
    "clevertap-react-native": "^2.2.1",
    "dayjs": "^1.11.7",
    "expo": "^51.0.26",
    "expo-application": "~5.9.1",
    "expo-checkbox": "~3.0.0",
    "expo-clipboard": "~6.0.3",
    "expo-constants": "~16.0.2",
    "expo-dev-client": "~4.0.22",
    "expo-device": "~6.0.2",
    "expo-font": "~12.0.9",
    "expo-image": "~1.12.13",
    "expo-linking": "~6.3.1",
    "expo-location": "~17.0.1",
    "expo-notifications": "~0.28.15",
    "expo-splash-screen": "~0.27.5",
    "expo-status-bar": "~1.12.1",
    "expo-updates": "~0.25.22",
    "geolib": "^3.3.3",
    "lottie-react-native": "6.7.0",
    "patch-package": "^8.0.0",
    "postinstall-postinstall": "^2.1.0",
    "prop-types": "^15.8.1",
    "react": "18.2.0",
    "react-hook-form": "^7.39.1",
    "react-native": "0.74.5",
    "react-native-big-list": "^1.6.0",
    "react-native-collapsible": "^1.6.0",
    "react-native-confirmation-code-field": "^7.3.1",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-gradle-plugin": "^0.71.19",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-maps": "1.14.0",
    "react-native-mask-input": "^1.2.2",
    "react-native-reanimated": "~3.10.1",
    "react-native-reanimated-carousel": "^3.1.5",
    "react-native-safe-area-context": "4.10.5",
    "react-native-screens": "3.31.1",
    "react-native-svg": "15.2.0",
    "react-native-toast-message": "^2.1.5",
    "react-native-webview": "13.8.6",
    "rn-iphone-helper": "^2.0.2",
    "valtio": "^1.7.5",
    "yup": "^1.4.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "babel-plugin-module-resolver": "^5.0.0",
    "eslint": "^9.9.0",
    "eslint-config-universe": "^13.0.0",
    "prettier": "^3.3.3"
  },
  "private": true
}

```

**Reply by MatiPl01** [reply]
Seems very strange. Would you be able to send me the complete call stack you get in the app after the crash? In my case, when I faked this crash, the real file that the error was thrown from was deep in the stack, so can you please record the screen and scroll down to show the whole stack or make a few screenshots? Maybe the culprit will be visible in there.

<img src="https://github.com/user-attachments/assets/b9fe610e-f9db-418d-8e71-6a73ca55ed65" width="300" />

**Reply by dafrica1** [reply]
The HeaderSection.js file is what is what we want to work. I can't find where the conflicting `opacity` whitelist is however.

![image](https://github.com/user-attachments/assets/3dc03410-23c5-4f39-aa94-e5cfa0a1fd5f)

**Reply by MatiPl01** [reply]
The call stack shows line 15 of the `HeaderSection.js` file. Can you share code showing what is in this line?

**Reply by dafrica1** [reply]
That's the line where I whitelist the native prop opacity:

`Animated.addWhitelistedNativeProps({ opacity: true });`

The issue is that I can't find where else in the app `opacity` is whitelisted as a UI prop

**Reply by MatiPl01** [reply]
Thanks for clarification. Now I can understand what is going on. 
It is caused by the fact that the `opacity` prop is whitelisted as a UI prop by default. Error is thrown in this function in the `react-native-reanimated` library, which this `key in PropsAllowlists.UI_THREAD_PROPS_WHITELIST` check.

```tsx
function assertNoOverlapInLists() {
  for (const key in PropsAllowlists.NATIVE_THREAD_PROPS_WHITELIST) {
    if (key in PropsAllowlists.UI_THREAD_PROPS_WHITELIST) {
      throw new Error(
        `[Reanimated] Property \`${key}\` was whitelisted both as UI and native prop. Please remove it from one of the lists.`
      );
    }
  }
}
```

As you can see in [this](https://github.com/software-mansion/react-native-reanimated/blob/b2f4e1dcfe51004becc0fb19d99ee073bf2c1229/packages/react-native-reanimated/src/propsAllowlists.ts#L11) file, `PropsAllowlists.UI_THREAD_PROPS_WHITELIST` contains `opacity`, so it will it is automatically whitelisted as a UI prop and you shouldn't change that.

I understand that the error might be confusing because you whitelist this prop only as a native prop but, since reanimated whitelists it automatically as the UI prop, you cannot do this.

**Reply by MatiPl01** [reply]
I wonder why you want this property to be whitelisted as a native prop instead of the UI prop? Is there any specific use case or you added this just by mistake?
